sql/src/main/java/io/crate/operation/collect/files/FileReadingCollector.java

```diff
@@ -202,6 +185,37 @@ public void doCollect(RamAccountingContext ramAccountingContext) {
         downstream.finish();
     }
 
+    private void readLines(FileInput fileInput, CollectorContext collectorContext, URI uri, long startLine, int retry) throws IOException {
+        InputStream inputStream = fileInput.getStream(uri);
+        if (inputStream == null) {
+            return;
+        }
+
+        String line;
+        long linesRead = 0L;
+        try (BufferedReader reader = createReader(inputStream)) {
+            while ((line = reader.readLine()) != null) {
+                linesRead++;
+                if (linesRead < startLine) {
+                    continue;
+                }
+                if (line.length() == 0) { // skip empty lines
+                    continue;
+                }
+                collectorContext.lineContext().rawSource(line.getBytes(StandardCharsets.UTF_8));
+                if (!downstream.setNextRow(row)) {
+                    break;
+                }
+            }
+        } catch (SocketTimeoutException e) {
+            if (retry > MAX_SOCKET_TIMEOUT_RETRIES) {
+                throw e;
+            } else {
+                readLines(fileInput, collectorContext, uri, linesRead + 1, retry + 1);
```

Comment:
hm afaik we must pass linesRead, not +1 as the current line was not read completely